Humanise() localization

Tkael opened the pull request:
Resolves #2067

Tkael committed:
Calculate our order multiplier automatically rather than explicitly setting it.

Tkael committed:
Calculating `number` and `nextDigit` may be removed from conditional statements.

Tkael committed:
Use the same coding format for describing numbers above and below one hundred.

Tkael committed:
Use long string phrases for localizing humanized values. Describe negative values with "negative" rather than "minus"

richardbuckle commented on Tests/SpeechUnitTests.cs:
```diff
@@ -279,7 +279,7 @@ public void TestSectorTranslations()
         public void TestSpeechHumanize1()
         {
-            Assert.AreEqual("well over minus 12 thousand", Translations.Humanize(-12345));
+            Assert.AreEqual("well over negative 12 thousand", Translations.Humanize(-12345));
```
Maybe this is a UK/US thing, but over here we would say "minus", at least in everyday speech.

Tkael replied:
I was thinking of negative as the opposite of positive, but upon reflection and after reviewing norms for describing temperatures both in the US and UK I have to agree. I've reverted from "negative" to "minus". o7

richardbuckle reviewed (approved):
Looks good code-wise. Note sure about saying "negative" instead of "minus" but I don't think it's worth adding an `en-GB` localisation for.

It might be a good idea to give the translators a heads-up on CrowdIn.

richardbuckle commented:
On reflection I can't think of many cases where EDDI has to speak negative numbers besides temperatures.

Tkael committed:
Prefer "minus" to "negative".
- negative numbers described in Elite are typically temperatures.
- "minus" is used more commonly than "negative" for describing numbers, particularly in the UK.
